pages/users/[id].tsx

```diff
@@ -251,24 +235,21 @@ export default function User({ loginContext }: Props) {
               {/* Recent Activity */}
               {$activeTab !== 'settings' && (
                 <>
-                  <h1 className="font-favorit">Recent Activity</h1>
+                  <h1 className="font-favorit" id="recent-activity">
+                    Recent Activity
+                  </h1>
 
                   <table className="font-favorit w-full">
                     <thead>
                       <tr className="text-xs text-left tracking-widest border-b border-black">
                         <th className="font-normal py-4">ACTIVITY</th>
                         <th className="font-normal">DATE</th>
-                        <th className="font-normal">POINTS EARNED</th>
+                        <th className="font-normal">POINTS</th>
+                        <th className="font-normal max-w-[13rem]">DETAILS</th>
                       </tr>
                     </thead>
                     <tbody className="text-sm">
-                      {$events.data.map(e => (
-                        <tr key={e.id} className="border-b border-black">
-                          <td className="py-4">{displayEventType(e.type)}</td>
-                          <td>{new Date(e.occurred_at).toLocaleString()}</td>
-                          <td>{e.points}</td>
-                        </tr>
-                      ))}
+                      {renderEvents(signupDate, $events.data as API.ApiEvent[])}
```

Comment:
Do you want to propagate the `readonly` or remove it here so it's not necessary to hard-cast it? That way we'll avoid having to update the cast if we change the type of `$events` in the future.

Reply:
TIL about this. Added to one function and then a different callsite needed this casting because it does a `.sort`.